homeassistant/components/yale_smart_alarm/alarm_control_panel.py

```diff
@@ -46,66 +60,106 @@
 )
 
 
-def setup_platform(hass, config, add_entities, discovery_info=None):
-    """Set up the alarm platform."""
-    name = config[CONF_NAME]
-    username = config[CONF_USERNAME]
-    password = config[CONF_PASSWORD]
-    area_id = config[CONF_AREA_ID]
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Import Yale configuration from YAML."""
+    LOGGER.warning(
+        "Loading Yale Alarm via platform setup is depreciated; Please remove it from your configuration"
+    )
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_IMPORT},
+            data=config,
+        )
+    )
+
 
-    try:
-        client = YaleSmartAlarmClient(username, password, area_id)
-    except AuthenticationError:
-        _LOGGER.error("Authentication failed. Check credentials")
-        return
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up the alarm entry."""
 
-    add_entities([YaleAlarmDevice(name, client)], True)
+    async_add_entities(
+        [YaleAlarmDevice(coordinator=hass.data[DOMAIN][entry.entry_id]["coordinator"])]
+    )
 
 
-class YaleAlarmDevice(AlarmControlPanelEntity):
+class YaleAlarmDevice(CoordinatorEntity, AlarmControlPanelEntity):
     """Represent a Yale Smart Alarm."""
 
-    def __init__(self, name, client):
-        """Initialize the Yale Alarm Device."""
-        self._name = name
-        self._client = client
-        self._state = None
+    _state_map = {
```

Comment:
This will stay static right? If so, we can make this a constant instead of a class attribute

Reply:
Have changed to a constant